### PR TITLE
Integration Tests for #8897 - Category `url_key` and `url_path` by scope

### LINKED ISSUE #29530: [Issue] Integration Tests for #8897 - Category `url_key` and `url_path` by scope
This issue is automatically created based on existing pull request: magento/magento2#28604: Integration Tests for #8897 - Category `url_key` and `url_path` by scope

---------
### Description (*)
During the review of my Closed PRs, I found out that there are still valuable changes, that should be added to Magento itself, to keep an eye of changes made to generating URLs for Categories.

:heavy_check_mark: Scenario 1: Change of `url_key` for a specific scope should not affect global scope
:heavy_check_mark: Scenario 2: Change of `url_key` for a specific scope should update `url_path` of children categories in specified scope

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
N/A

### Questions or comments
N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
During the review of my Closed PRs, I found out that there are still valuable changes, that should be added to Magento itself, to keep an eye of changes made to generating URLs for Categories.

:heavy_check_mark: Scenario 1: Change of `url_key` for a specific scope should not affect global scope
:heavy_check_mark: Scenario 2: Change of `url_key` for a specific scope should update `url_path` of children categories in specified scope

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
N/A

### Questions or comments
N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29530: Integration Tests for #8897 - Category `url_key` and `url_path` by scope

### CODE REVIEW

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/Catalog/Controller/Adminhtml/Category/Save/SaveCategoryDifferentScopesTest.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    {
+        parent::setUp();
+        $this->categoryRepository = $this->_objectManager->get(CategoryRepositoryInterface::class);
+        $this->storeManager = $this->_objectManager->get(StoreManagerInterface::class);
+    }
+
+    /**
+     * @magentoDataFixture Magento/Catalog/_files/category.php
+     */
+    public function testUpdateUrlKeyForStoreExpectNoChangeOfGlobalUrlPath()
```
Comment (rogyar) [inline dev/tests/integration/testsuite/Magento/Catalog/Controller/Adminhtml/Category/Save/SaveCategoryDifferentScopesTest.php:49]: Is this one supposed to be failing for the current core functionality?

Comment (lbajsarowicz) [reply dev/tests/integration/testsuite/Magento/Catalog/Controller/Adminhtml/Category/Save/SaveCategoryDifferentScopesTest.php:49]: That is the tricky part. This should not fail, but I also noticed it is failing, that means that in the last 3 months, that feature was broken. I'm going to investigate and fix this in the scope of this PR.

**Comment by rogyar** [comment]
@magento run Integration Tests

**Comment by sidolov** [comment]
@magento create issue

File: dev/tests/integration/testsuite/Magento/Catalog/Controller/Adminhtml/Category/Save/SaveCategoryDifferentScopesTest.php (hunk 1 above)
Comment (sidolov) [reply dev/tests/integration/testsuite/Magento/Catalog/Controller/Adminhtml/Category/Save/SaveCategoryDifferentScopesTest.php:49]: @lbajsarowicz do you had a chance to investigate the issue?

Comment (lbajsarowicz) [reply dev/tests/integration/testsuite/Magento/Catalog/Controller/Adminhtml/Category/Save/SaveCategoryDifferentScopesTest.php:49]: Let me merge the latest `2.4-develop` branch and run that once again.
I'm pretty sure I know which exactly PR broke that, but I'll handle that at the beginning of September unless you arrange someone else.

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @lbajsarowicz,
Seems like the test is wrong OR the fix was reverted. Could you check it?

**Comment by lbajsarowicz** [comment]
I can't get some spare time to review that.
Let's close that for now.

**Pull request closed without merging**